plugins/modules/hostgroup.py

```diff
@@ -200,6 +200,20 @@ def main():
         ansible_roles = module_params.get('ansible_roles')
         if not module.desired_absent and ansible_roles is not None:
             desired_ansible_role_ids = [item['id'] for item in ansible_roles]
+
+            if entity['ancestry'] is not None:
```

Comment:
```suggestion
            if entity.get('ancestry') is not None:
```

(for some reason it's sometimes not defined at all)

Reply:
Yes, you are absolutely right.

I also noticed that `parent_id` is defined more consistently (even in check mode).
If you don't see any problems with that, I would use `entity.get('parent_id')` instead of `entity.get('ancestry')`.

Reply:
I never looked much into the whole ancestry stuff, so I'll just go and trust you here :)